src/main/java/camera/Camera.java

```diff
@@ -13,51 +17,142 @@
      * Constructor for a Camera.
+     *
      * @param newCapture The VideoCapture of this camera.
-     * @param newLink The link of this camera.
+     * @param newLink    The link of this camera.
      */
     public Camera(final VideoCapture newCapture, final String newLink) {
+        final int threshold = 1000;
         cameraObjectDetectorList = new ArrayList<>();
         videoCapture = newCapture;
         link = newLink;
+        activity = new ArrayList<>();
+        for (int i = 0; i < frames; i++) {
+            frameParts.add(new Mat());
+            activity.add(new ArrayList<>());
+            knns.add(Video.createBackgroundSubtractorKNN(1, threshold, false));
+        }
+        activity.add(new ArrayList<>());
+        knns.add(Video.createBackgroundSubtractorKNN(1, threshold, false));
     }
 
     /**
      * Gets the last known frame of this camera.
+     *
      * @return The frame in Mat format.
      */
     public Mat getLastFrame() {
+        final int modulus = 5;
         Mat newFrame = loadFrame();
 
         if (newFrame != null && newFrame.rows() != 0 && newFrame.cols() != 0) {
             lastFrame = newFrame;
             changed = true;
         } else {
             changed = false;
+            return lastFrame.clone();
+        }
+
+        divideFrame(newFrame);
+
+        if (frameCounter % modulus == 0) {
+            for (int i = 0; i < frames; i++) {
+                addActivity(frameParts.get(i), i, knns.get(i));
+            }
         }
+        addActivity(newFrame, frames, knns.get(frames));
+        frameCounter++;
 
         return lastFrame.clone();
     }
 
+    /**
+     * Method in which a frame get divided into frames equally large parts.
+     * frames should be a number where the sqrt is an integer
+     *
+     * @param frame the input frame
+     */
+    public void divideFrame(final Mat frame) {
+        for (int i = 0; i < frames; i++) {
+            int sqrt = (int) Math.sqrt(frames);
+            int midCol = frame.width() / sqrt;
+            int midRow = frame.height() / sqrt;
+
+            int col = midCol * (i % sqrt);
+            int row = (i * sqrt) / frames * midRow;
+
+            Mat result = frame.colRange(col, col + midCol);
+            result = result.rowRange(row, row + midRow);
+            frameParts.set(i, result);
+        }
+    }
+
+    /**
+     * Add current activity to list.
+     * @param frame the current frame
+     * @param i current part of the frame
+     * @param knn the backgroundsubstractor
+     */
+    public void addActivity(final Mat frame, final int i, final BackgroundSubtractorKNN knn) {
+        Mat subtraction = new Mat();
+        knn.apply(frame, subtraction);
+        Scalar meanValues = Core.mean(subtraction);
+
+        double change = 0;
+        for (double v : meanValues.val) {
+            change += v;
+        }
+
+        final int minFrames = 30;
+        final double timeWindow = 1000.00;
+        if (frameCounter > minFrames) {
```

Comment:
Maybe add a comment here about what (frameCounter > minFrames) exactly means. I think it means that there will be activity added to the list only when "enough" frames have passed but I'm not entirely sure, some clarification would be nice.